jib-core/src/main/java/com/google/cloud/tools/jib/api/DockerDaemonImage.java

```diff
@@ -17,20 +17,12 @@
 package com.google.cloud.tools.jib.api;
 // TODO: Move to com.google.cloud.tools.jib once that package is cleaned up.
 
-import com.google.cloud.tools.jib.builder.BuildSteps;
-import com.google.cloud.tools.jib.configuration.BuildConfiguration;
-import com.google.cloud.tools.jib.configuration.ImageConfiguration;
-import com.google.cloud.tools.jib.docker.DockerClient;
 import com.google.cloud.tools.jib.image.ImageReference;
 import com.google.cloud.tools.jib.image.InvalidImageReferenceException;
-import com.google.common.collect.ImmutableMap;
 import java.nio.file.Path;
 import java.util.Map;
-import javax.annotation.Nullable;
 
-/** Builds to the Docker daemon. */
-// TODO: Add tests once JibContainerBuilder#containerize() is added.
-public class DockerDaemonImage implements TargetImage {
+public interface DockerDaemonImage {
```

Comment:
So I kinda like this change except that it would be more in keeping if `DockerDaemonImage` used the builder pattern as used with `TarImage`.  Then we could avoid the oddity of the object supporting `set*` methods.
```java
DockerDaemonImage.builder("image/ref")
  .setDockerEnvironment(map...)
  .build();
```

Reply:
Unrelated to this: looking at `TarImage` again, I think the `Builder` is not a right choice as a name used for the fluent interface pattern.